general/argo-workflow-gc/template.yaml

```
@@ -0,0 +1,27 @@
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: argo-workflow-enforce-podgc-field
```

Not sure if we want to align the name as close to the `Kind`s name?
```suggestion
  name: argo-workflow-enforce-podgc-field
```